unlock-app/app/frames/checkout/components/utils.ts

```diff
@@ -42,6 +73,16 @@ export async function getLockDataFromCheckout(id: string) {
   const res = await web3Service.getLock(lockAddress, network)
   const price = `${res.keyPrice} ${res.currencySymbol}`
 
+  let tokenAddress
+  if (res.currencySymbol !== 'ETH') {
```

Comment:
That's not the right way to check if the lock is priced in the native currency. Thus would fail on Polygon for example. Just check that the `lock.tokenAddress` is the Zero address.